src/cc/libbpf.c

```diff
@@ -796,15 +797,18 @@ static int bpf_try_perf_event_open_with_probe(const char *name, uint64_t offs,
   /* config1 here is kprobe_func or  uprobe_path */
   attr.config1 = ptr_to_u64((void *)name);
   // PID filter is only possible for uprobe events.
-  if (pid < 0)
+  if (cgroup_fd >= 0) {
```

Comment:
I think we should also warn (or even error) if user passed in both PID and cgroup FD, which is possible in current interface.